
Patch to fix compilation with gcc-4.3, thanks to Cyril Brulebois.

--- a/GIFTServer/modifyDistanceMatrix.cc
+++ b/GIFTServer/modifyDistanceMatrix.cc
@@ -15,6 +15,7 @@
 #include <string>
 #include <map>
 #include <functional>
+#include <cstdlib>
 
 using namespace std;
 
--- a/libGIFTAcDistanceMatrix/include/CPersistentTranslatedIndexMatrix.h
+++ b/libGIFTAcDistanceMatrix/include/CPersistentTranslatedIndexMatrix.h
@@ -75,8 +75,8 @@
   bool getLineVector(const TTID& inI,
 		     CIDContentPairList& outList)const;
   ///
-  bool setValue(const TTID& inX,
-		const TTID& inX,
+  bool setValue(const TTID& inX1,
+		const TTID& inX2,
 		const TTContent& inValue)const;
   ///
   operator bool()const;
--- a/libGIFTQuInvertedFile/cc/CWeightingFunction.cc
+++ b/libGIFTQuInvertedFile/cc/CWeightingFunction.cc
@@ -17,7 +17,7 @@
 ***************************************************/
 #include <cmath>
 #include "libGIFTQuInvertedFile/include/CWeightingFunction.h"
-#include "iostream.h"
+#include <iostream>
 #include "FeatureExtraction/gift_features.h"
 
 #include <algorithm>
--- a/libMRML/cc/CAccessorImplementation.cc
+++ b/libMRML/cc/CAccessorImplementation.cc
@@ -54,6 +54,7 @@
 #include <iostream>
 #include <cassert>
 #include <cmath>
+#include <cstdlib>
 
 #define _NO_PRINT_OFFSET_CHECK
 #define _NO_CHECK_OFFSET_FILE
--- a/libMRML/cc/CCommunicationHandler.cc
+++ b/libMRML/cc/CCommunicationHandler.cc
@@ -16,6 +16,7 @@
 #include <stdlib.h>
 #include "libMRML/include/mrml_const.h" //mrml string constants newStart/EndMRMLElement
 #include "../include/CTimeStampGenerator.h" // for time stamps
+#include <cstring>
 string gGIFTHome;//dirty but necessary
 CMutex* gMutex(0);   //we need a global one
 
--- a/libMRML/cc/CDynamicQueryFactory.cc
+++ b/libMRML/cc/CDynamicQueryFactory.cc
@@ -31,6 +31,7 @@
 // for file name treatment
 #include "libMRML/include/getLibNameFromFileName.h"
 #include <set>
+#include <cstdlib>
 
 CQuery* CDynamicQueryFactory::makeQuery(const string & inBaseType, 
 					   CAccessorAdminCollection & inAccessorAdminCollection,
--- a/libMRML/cc/CI18nTranslator.cc
+++ b/libMRML/cc/CI18nTranslator.cc
@@ -25,6 +25,7 @@
 #include "libMRML/include/CXEVI18nTranslator.h"
 #include <iostream>
 #include <fstream>
+#include <cstring>
 /**
    This class offers runtime translation of strings. In contrast
    to GNU gettext we can chose during runtime the language without 
--- a/libMRML/cc/CQueryTreeBuilder.cc
+++ b/libMRML/cc/CQueryTreeBuilder.cc
@@ -3,6 +3,7 @@
 #include "libMRML/include/CQueryTreeBuilder.h"
 #include "libMRML/include/CAlgorithm.h"
 #include "libMRML/include/mrml_const.h"
+#include <cstring>
 //----------------------------------------
 
 CQueryTreeBuilder::CQueryTreeBuilder(){};
--- a/libMRML/cc/CXMLElementBuilder.cc
+++ b/libMRML/cc/CXMLElementBuilder.cc
@@ -1,6 +1,7 @@
 #include <memory>
 #include <iostream> // for printouts
 #include "libMRML/include/CXMLElementBuilder.h"
+#include <cstring>
 
 void newStartXMLElement(void *inUserData, 
 			 const char *inElementName, 

